pages/api/group/[gid]/index.ts

```diff
@@ -0,0 +1,69 @@
+import mongoose from 'mongoose';
+import type { NextApiRequest, NextApiResponse } from 'next';
+import {dbConnect} from '@/lib/mongoClient';
+import { responseFormatter } from '@/lib/responseLib';
+import Group from '@/models/Group';
+import  UserGroupRelation  from '@/models/UserGroupRelation';
+
+export default async function gidHandler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+
+  // Connect with MongoDB database
+  try {
+    await dbConnect();
+  } catch (e) {
+    console.log('fail to connect with database', e);
+    return (
+      res
+        .status(500)
+        .json(responseFormatter(false, null, 'Internal Server Error'))
+    );
+  }
+  const {gid} = req.query
+  const group = await Group.findOne({gid: gid});
+
+  switch (req.method) {
+    case 'POST':
+      try {
+        //update group details
+        const groupData = req.body;
+        await Group.findOneAndUpdate({gid:gid}, {...groupData});      
```

Comment:
Again, you would like to do some schema validation. The classical quote, a backend developer should never trust the front end data